src/InsertionsClient/Api/DefaultConfigUpdater.cs

```diff
@@ -197,6 +197,14 @@ public FileSaveResult[] Save()
 
                     using XmlWriter writer = XmlWriter.Create(savePath, writeSettings);
                     document.Save(writer);
+                    writer.Close();
```

Comment:
Can we add the new line before closing the stream? I think we already know that XmlWriter doesn't create a new line at the end. We can add it without checking. Something like this:
```
Stream fileStream = File.OpenRead(path);
var writer = XmlWriter.Create(fileStream, writeSettings);
document.Save(writer);
if(path.extension == ".config")
{
    fileStream.Write(newLine);
}
writer.Close(); 
```

Currently, we close the file, read it all back to memory to check 1 character at the end. I though we can do it a bit more efficiently. But I'm not sure if it is worth our time, so your call :)

Reply:
The reason:
Without `writer.Close()` you'd have some cases where the file is locked & cannot be accessed for the potential "new line" append.  Depends on how fast the handle is collected after disposal.

I can time box time to search for an efficient way to read the last line of a file.

Reply:
@bozturkMSFT exceeded my time box & didn't identify a much better way to tail the file to test the last line within the project.
We could blindly append a new line, since XDoument.ToString() rids of blank lines in the original file it loaded.  In my opinion the check makes the append more costly, but deterministic.